app/controllers/IndexController.scala

```diff
@@ -50,14 +50,23 @@ class IndexController @Inject()(
           case Some(userAnswers) if userAnswers.get(AcknowledgementReferencePage).isDefined =>
             Future.successful(Redirect(routes.RegistrationSentController.onPageLoad()))
           case _ =>
+            val cacheData = charitiesKeyStoreService.getCacheData(request)
+
+
             for {
-              userAnswers <- charitiesKeyStoreService.getCacheData(request)
-              _ <- userAnswerRepository.set(userAnswers)
+              pastUserAnswersWithErrors <- cacheData
+              _ <- userAnswerRepository.set(pastUserAnswersWithErrors._1)
               isSwitchOver <- cache.fetchAndGetEntry[Boolean](sessionId.value, IsSwitchOverUserPage)
             } yield {
-              val result = taskListHelper.getTaskListRow(userAnswers)
-              val completed = result.reverse.tail.forall(_.state.equals("index.section.completed"))
-              Ok(view(result, status = completed, isSwitchOver))
+              pastUserAnswersWithErrors._2 match {
```

Comment:
it would be good to have named tuples like
(userAnswers, validationErrors) <- charitiesKeyStoreService.getCacheData(request)

Reply:
on it